src/components/CardType/index.tsx

```diff
@@ -0,0 +1,35 @@
+import { useContext, useEffect, useState } from 'react'
+import { UpdateCardInfoContext } from '../../context/paymentContext'
+import ui from '../../styles/index.module.css'
+
+export const CardType = ({ setIsShowModal }: { setIsShowModal: () => void }) => {
+  const updateCardInfo = useContext(UpdateCardInfoContext)
+  const [selectType, setSelectType] = useState('')
+
+  useEffect(() => {
+    if (selectType) updateCardInfo({ cardType: selectType })
+  }, [selectType])
+
+  return (
+    <div className={ui['modal-dimmed']} onClick={() => setIsShowModal()}>
```

Comment:
위에도 언급드린 부분이긴한데, set 네이밍을 가진 setIsShowModal을 인수 없이 호출했을때 자연스럽게 어색하다고 느끼신다면 좋을 것 같아요!

```suggestion
    <div className={ui['modal-dimmed']} onClick={closeModal}>
```